src/clientdlg.cpp

```diff
@@ -1031,6 +1035,34 @@ void CClientDlg::OnTimerSigMet()
     // show current level
     lbrInputLevelL->SetValue ( pClient->GetLevelForMeterdBLeft() );
     lbrInputLevelR->SetValue ( pClient->GetLevelForMeterdBRight() );
+
+    if ( bDetectFeedback &&
+        ( pClient->GetLevelForMeterdBLeft() > NUM_STEPS_LED_BAR - 0.5  ||
+        pClient->GetLevelForMeterdBRight() > NUM_STEPS_LED_BAR - 0.5 ) )
+    {
+        // mute locally and mute channel
+        chbLocalMute->setCheckState ( Qt::Checked );
+        MainMixerBoard->MuteMyChannel();
+
+        // show message box about feedback issue
+        QCheckBox* chb = new QCheckBox ( tr ( "Enable feedback detection" ) );
+        chb->setCheckState ( pSettings->bEnableFeedbackDetection ?
+            Qt::Checked : Qt::Unchecked );
+        QMessageBox msgbox;
+        msgbox.setText ( tr ( "Audio feedback or loud signal detected.\n\n"
+            "We muted your channel and activated 'Mute Myself'. Please "
+            "solve the feedback issue first and unmute yourself afterwards. "
+            "Did you plug-in earphones?") );
```

Comment:
As the feedback detection only triggers upon connect, should we advise users to disconnect and try again here?

Reply:
I would not count on unexperienced users disconnecting and connecting again. Although it would achieve the goal, it would not be particularly intuitive.

Initially I thought about re-triggering the feedback detection interval after un-muting -- that would be a reliable technical solution.

Reply:
I'm not saying we have to do it at all, I was just thinking that adjusting the text might direct users in a good direction without any more technical effort.
Re-triggering the feedback detection after un-mutung would also work for me but would need additional logic (even if not much).